string.c

```diff
@@ -1556,21 +1556,82 @@ rb_str_resurrect(VALUE str)
 
 /*
  *  call-seq:
- *     String.new(str="")                   -> new_str
- *     String.new(str="", encoding: enc)    -> new_str
- *     String.new(str="", capacity: size)   -> new_str
- *
- *  Returns a new string object containing a copy of <i>str</i>.
- *
- *  The optional <i>encoding</i> keyword argument specifies the encoding
- *  of the new string.
- *  If not specified, the encoding of <i>str</i> is used
- *  (or ASCII-8BIT, if <i>str</i> is not specified).
- *
- *  The optional <i>capacity</i> keyword argument specifies the size
- *  of the internal buffer.
- *  This may improve performance, when the string will be concatenated many
- *  times (causing many realloc calls).
+ *     String.new(str='')                   -> new_str
+ *     String.new(str='', encoding: enc)    -> new_str
+ *     String.new(str='', capacity: size)   -> new_str
+ *
+ *  If argument +str+ is given, it must be a
+ *  {String-convertible object}[doc/implicit_conversion_rdoc.html#label-String-Convertible+Objects]
+ *  (implements +to_str).
+ *
+ *  Argument +encoding+, if given, must be:
+ *  - A {String-convertible object}[doc/implicit_conversion_rdoc.html#label-String-Convertible+Objects]
+ *    (implements +to_str+).
+ *  - The name of an encoding that is compatible with +str+.
+ *
+ *  Argument +capacity+, if given, must be an
+ *  {Integer-convertible object}[doc/implicit_conversion_rdoc.html#label-Integer-Convertible+Objects]
+ *  (implements +to_int+).
+ *
+ *  Returns a new \String that is a copy of <i>str</i>.
+ *
+ *  ---
+ *
+ *  With no arguments, returns the empty string with encoding <tt>Encoding:ASCII-8BIT</tt>:
+ *    s = Str.new
+ *    s # => ""
+ *    s.encoding # => #<Encoding:ASCII-8BIT>
+ *
+ *  With the single argument +str+, returns a copy of +str+
+ *  with the same encoding as +str+:
+ *    s = String.new('Que veut dire ça?')
+ *    s # => "Que veut dire ça?"
+ *    s.encoding # => #<Encoding:UTF-8>
+ *
```

Comment:
There should be some note in this section that literal strings (`""`, `%q{}`, heredocs) use [script encoding](https://docs.ruby-lang.org/en/2.7.0/Encoding.html#class-Encoding-label-Script+encoding) while `String.new` defaults to Encoding::ASCII-8BIT.

Perhaps:

```rdoc
Literal strings like <tt>""</tt>, <tt>%q{}</tt> or here-documents always use {script encoding}[Encoding@Script+encoding] unlike String.new.
```

(Please check that my link works 🙇🏻‍♂️)

Reply:
Done, except that <tt>%q{}</tt> did not work.  Did not exit the tt font  properly.  I tried <code> and escaping:  no joy.  If you know how to fix and include, please do so.

Reply:
Oh, and the link now works.

Reply:
Ah, right, `++` only works for more-or-less ruby identifiers, sorry!